src/libopensc/card.c

```diff
@@ -899,19 +925,17 @@ int sc_get_challenge(sc_card_t *card, u8 *rnd, size_t len)
 	if (r != SC_SUCCESS)
 		LOG_FUNC_RETURN(card->ctx, r);
 
-	while (len > 0 && retry > 0) {
+	while (len > 0) {
 		r = card->ops->get_challenge(card, rnd, len);
+		if (r == 0)
+			r = SC_ERROR_INVALID_DATA;
 		if (r < 0) {
 			sc_unlock(card);
 			LOG_FUNC_RETURN(card->ctx, r);
 		}
 
-		if (r > 0) {
-			rnd += (size_t) r;
-			len -= (size_t) r;
-		} else {
-			retry--;
-		}
+		rnd += (size_t) r;
+		len -= (size_t) r;
```

Comment:
clang-format suggested the following change:


```
-		rnd += (size_t) r;
-		len -= (size_t) r;
+		rnd += (size_t)r;
+		len -= (size_t)r;
```

<!-- identifier: clang-format -->
<!-- File: src/libopensc/card.c -->